React/Views/RCTNavItem.m

```diff
@@ -11,10 +11,21 @@
 
 @implementation RCTNavItem
 
+@synthesize titleImageView = _titleImageView;
 @synthesize backButtonItem = _backButtonItem;
 @synthesize leftButtonItem = _leftButtonItem;
 @synthesize rightButtonItem = _rightButtonItem;
 
+- (UIImageView *)titleImageView
+{
+  if (_titleImage) {
+    _titleImageView = [[UIImageView alloc] initWithImage:_titleImage];
```

Comment:
I'm in favour of removing the `@synthesize`, comparing `UIImage`s will be more expensive. Will update the PR later today.